code/datums/components/unbreakable.dm

```diff
@@ -0,0 +1,31 @@
+/datum/component/unbreakable
+	COOLDOWN_DECLARE(surge_cooldown)
+
+/datum/component/unbreakable/Initialize()
+	if(!ishuman(parent))
+		return COMPONENT_INCOMPATIBLE
+	ADD_TRAIT(parent, TRAIT_UNBREAKABLE, INNATE_TRAIT)
+
+/datum/component/unbreakable/Destroy(force, silent)
+	. = ..()
+	REMOVE_TRAIT(parent, TRAIT_UNBREAKABLE, INNATE_TRAIT)
+
+/datum/component/unbreakable/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOB_STATCHANGE, .proc/surge)
+
+/datum/component/unbreakable/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_MOB_STATCHANGE)
+
+/datum/component/unbreakable/proc/surge(mob/living/carbon/human/surged, new_stat)
+	SIGNAL_HANDLER
+	if(new_stat != UNCONSCIOUS)
+		return
+	if(!COOLDOWN_FINISHED(src, surge_cooldown))
+		return
+	COOLDOWN_START(src, surge_cooldown, 1 MINUTES)
+	surged.balloon_alert(surged, "you refuse to give up!")//breaks balloon alert conventions by using a "!" for a fail message but that's okay because it's a pretty awesome moment
+	surged.heal_overall_damage(15, 15, 0, BODYPART_ORGANIC)
+	if(surged.reagents.get_reagent_amount(/datum/reagent/medicine/ephedrine) < 20)
+		surged.reagents.add_reagent(/datum/reagent/medicine/ephedrine, 10)
```

Comment:
are you intended to still get the ephedrine here?

also, is this 100% pure ephedrine, or does it carry a risk of you dropping what you're holding?

Reply:
1. yea
2. i'm pretty sure it's pure